src/Coin.java

```diff
@@ -1,22 +1,25 @@
 import javafx.scene.canvas.GraphicsContext;
 
-public class Coin extends Character {
+public class Coin extends WorldItem {
     private static final String _PATH = "./assets/coin.png";
-
     private BackgroundItem background;
     private int MOVING_SPEED;
+    private Constants constants;
 
     public Coin(BackgroundItem background, int MOVING_SPEED) {
         this.background = background;
         this.MOVING_SPEED = MOVING_SPEED;
         this.setImage(_PATH);
         this.resetPosition();
+        this.constants = new Constants(background);
```

Comment:
Constants doesn's appear to be used in the coin class and moving speed is still there and being used. Change this to wors